emily/lambda/src/resources/deposits.rs

```diff
@@ -0,0 +1,487 @@
+    /// Bitcoin fee used to fulfill the deposit.
+    pub btc_fee: u64,
+}
+
+// Conversions ------------------------------------------------------
+
+/// Convert from resource definition of deposit parameters to the api model version
+/// of the struct.
+impl From<DepositParameters> for Box<models::DepositParameters> {
+    fn from(value: DepositParameters) -> Self {
+        Box::new(models::DepositParameters {
+            max_fee: value.max_fee as f64,
+            lock_time: value.lock_time as f64,
+            reclaim_script: value.reclaim_script,
+        })
+    }
+}
+
+// ------------------------------------------------------------------
+// DynamoDB table entry resources.
+// ------------------------------------------------------------------
+
+/// Deposit table entry key.
+#[derive(Debug)]
+pub struct DepositTableEntryKey {
+    /// Bitcoin transaction id.
+    pub bitcoin_txid: String,
+    /// Bitcoin transaction output index.
+    pub bitcoin_tx_output_index: u16,
+}
+
+/// Deposit table entry.
+#[derive(Debug)]
+pub struct DepositTableEntry {
+    /// Bitcoin transaction id.
+    pub bitcoin_txid: String,
+    /// Bitcoin transaction output index.
+    pub bitcoin_tx_output_index: u16,
+    /// Table entry version. Updated on each alteration.
+    pub version: u64,
+    /// Recipient of the deposit.
+    pub recipient: String,
+    /// Amount of the deposit.
+    pub amount: u64,
+    /// Max fee in satoshis to execute the deposit.
+    pub max_fee: u64,
+    /// Lock time.
+    pub lock_time: u64,
+    /// Reclaim script.
+    pub reclaim_script: String,
+    /// Operation status.
+    pub op_status: models::OpStatus,
+    /// Last update block height.
+    pub last_update_height: u64,
+    /// Last update block hash.
+    pub last_update_block_hash: String,
+    /// Bitcoin transaction id that fulfills the deposit.
+    pub fulfillment_bitcoin_txid: Option<String>,
+    /// Output index on the fulfilling Bitcoin transaction that fulfills the deposit.
+    pub fulfillment_bitcoin_tx_index: Option<u16>,
+    /// Fee on the fulfillment bitcoin transaction paid by this deposit.
+    pub fulfillment_btc_fee: Option<u64>,
+    /// Stacks transaction id.
+    pub stacks_txid: Option<String>,
+    /// History of this deposit transaction.
+    pub history: Vec<DepositHistoryEntry>,
+}
+
+/// Deposit history entry.
+#[derive(Debug)]
+pub struct DepositHistoryEntry {
+    /// Status code.
+    pub op_status: models::OpStatus,
+    /// Status message.
+    pub message: String,
+    /// Stacks block heigh at the time of this update.
+    pub stacks_block_height: u64,
+    /// Stacks block hash associated with the height of this update.
+    pub stacks_block_hash: String,
+}
+
+// Conversions ------------------------------------------------------
+
+/// Converts from the deposit key to a hashmap that can be used as a key
+/// for searching a DynamoDB table.
+impl From<DepositTableEntryKey> for HashMap<String, AttributeValue> {
```

Comment:
I don't think this is necessary for any of these types. Instead, derive `Serialize` and `Deserialize` for the struct and just use `serde_dynamo::to_item` to transform the type to a `HashMap<String, AttributeValue>`. I think that will significantly simplify the code here.